bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+            name="How do I *delete* my spookified name?",
+            value=f"Simply type `{self.bot.command_prefix}spooknamerate delete`",
+            inline=False
+        )
+
+        await ctx.send(embed=help_embed)
+
+    @spook_name_rate.command(name='list', aliases=["all", "entries"])
+    async def list_entries(self, ctx: Context) -> None:
+        """Send all the entries up till now in a single embed."""
+        await ctx.send(embed=await self.get_responses_list(final=False))
+
+    @spook_name_rate.command(name='name')
+    async def tell_name(self, ctx: Context) -> None:
+        """Tell the current random name (name)."""
+        if not self.poll:
+            await ctx.send(f"The name is **{self.name}**")
+            return
+
+        await ctx.send(
+            f"The name ~~is~~ was **{self.name}**. The poll has already started, so you cannot "
+            "add an entry."
+        )
+
+    @spook_name_rate.command(name="add", aliases=["+", "register"])
+    async def add_name(self, ctx: Context, *, name: str) -> None:
+        """Use this command to add/register your spookified name."""
+        if self.poll:
+            logger.info(f"{ctx.message.author} tried to add a name, but the poll had already started.")
+            await ctx.send("Sorry, the poll has started! You can try and participate in the next round though!")
+            return
+
+        message = ctx.message  # get the message
+
+        for data in [json.loads(x[1]) for x in await self.messages.items()]:
+            if data['author'] == message.author.id:  # if the author has already added an entry
+                await ctx.send(
+                    "But you have already added an entry! Type "
+                    f"`{self.bot.command_prefix}spooknamerate "
+                    "delete` to delete it, and then you can add it again"
+                )
+                return
+
+            elif data['name'] == name:
+                # Someone has added the name
+                await ctx.send("TOO LATE. Someone has already added this name.")
+                return
+
+        await self.messages.set(message.id, json.dumps({  # store it
+            'name': name,
+            'author': message.author.id,
+            'score': 0,
+        }))
+
+        for emoji in EMOJIS_VAL:
+            await message.add_reaction(emoji)  # add each emoji
+
+        logger.info(f"{message.author} added the name {name!r}")
+
+        await ctx.send(f"{name!r} added successfully!\n{random.choice(self.added_messages)}")
+
+    @Cog.listener()
+    async def on_reaction_add(self, reaction: Reaction, user: User) -> None:
+        """Ensures that each user adds one and only one reaction."""
+        async with self.checking_messages:  # Acquire the lock so that the dictionary isn't reset while iterating.
+            if reaction.emoji in EMOJIS_VAL and await self.messages.contains(reaction.message.id):
+                msg: Message = reaction.message
```

Comment:
We usually don't type hint variables.

Reply:
Oh I was just using that for intellisense. I'll remove it.